db/db.go

```diff
@@ -3141,3 +3141,78 @@ func GetSyncParticipationBySlotRange(startSlot, endSlot uint64) (map[uint64]uint
 
 	return ret, nil
 }
+
+// Should be used when retrieving data for a very large amount of validators (for the notifications process)
+func GetValidatorAttestationHistoryForNotifications(validators []uint64, startEpoch uint64, endEpoch uint64) (map[types.Epoch]map[types.ValidatorIndex]bool, error) {
```

Comment:
**Bug:** Passing only a single validator will still give attestations for all validators in the epoch range as all attestations are added here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/31375fc09d88f2c733a14c315565ce646809dc92/db/db.go#L3212-L3214